core/src/main/java/com/flipkart/gap/usl/core/client/KafkaClient.java

```diff
@@ -50,7 +51,16 @@ public void init() {
         consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer");
         consumer = new KafkaConsumer<>(consumerProperties);
         producer = new KafkaProducer<>(props);
-        producerCount = producer.partitionsFor(eventProcessorConfig.getTopicName()).size();
+        if (StringUtils.isNotEmpty(eventProcessorConfig.getTopicName()))
```

Comment:
why this change? and in case this parameter is not there we are ignoring it silently, would be difficult to debug